File: src/lesson011/java/CarTest.java
package lesson011.java;

public class CarTest {

    public static void main(String[] args) {
        Car car1 = new Car(4, "red", "v8");
        Car car2 = new Car(3, "blue", "v4");
        Car car3 = new Car(5, "green", "v6");

        showInfo(car1);
        showInfo(car2);
        showInfo(car3);
        System.out.println("-------------------------------------------");
        changeDoors(car1, 2);
        swapColors(car2, car3);
        System.out.println("-------------------------------------------");
        showInfo(car1);
        showInfo(car2);
        showInfo(car3);
    }

    static void showInfo(Car car){
        System.out.println("Car: doors: "+car.doors+", color: "+car.color+", engine: "+car.engine+";");
    }


    static void changeDoors(Car car, int doors){
        System.out.println("Before doors: "+car.doors);
        car.doors = doors;
        System.out.println("Now doors: "+car.doors);
    }

    static void swapColors(Car car1, Car car2){
        String colorTemp = car1.color;
        car1.color = car2.color;
        car2.color = colorTemp;
    }
}

class Car{
    public int doors;
    public String color;
    public String engine;

    public Car(){}

    public Car(int doors, String color, String engine) {
        this.doors = doors;
        this.color = color;
        this.engine = engine;
    }
}

File: src/lesson011/kotlin/CarTest.kt
package lesson011.kotlin

fun main(args: Array<String>) {
    val car1 = Car(4, "red", "V8")
    val car2 = Car(3, "blue", "V4")
    val car3 = Car(5, "green", "V6")

    fun showInfo(car: Car) {
        println("Car doors: ${car.doors}, color: ${car.color}, engine: ${car.engine};")
    }

    fun changeDoors(car: Car, doors: Int){
        car.doors = doors
    }

    fun swapColor(car1: Car, car2: Car){
        val colorTemp = car1.color
        car1.color = car2.color
        car2.color = colorTemp
    }

    showInfo(car1)
    showInfo(car2)
    showInfo(car3)
    println("-------------------------------------")
    changeDoors(car1, 5)
    swapColor(car2, car3)
    showInfo(car1)
    showInfo(car2)
    showInfo(car3)

}

class Car(){
    var doors = 4
    var color = "white"
    var engine = "v4"

    constructor(doors: Int, color: String, engine: String): this(){
        this.doors = doors
        this.color = color
        this.engine = engine
    }
}